scan/lib/scan/detect_values.rb

```diff
@@ -187,15 +187,16 @@ def self.detect_simulator(devices, requested_os_type, deployment_target_key, def
         end
       end
 
-      # grab the first unempty evaluated array
-      if default
-        Scan.devices = [matches, default].lazy.map { |x|
-          arr = x.call
-          arr unless arr.empty?
-        }.reject(&:nil?).first
-      else
-        Scan.devices = []
+      # default to an empty array if `skip_detect_devices` is `true`
+      unless default
+        default = []
       end
+
+      # grab the first unempty evaluated array
+      Scan.devices = [matches, default].lazy.map { |x|
+        arr = x.call
```

Comment:
I tried running with this an unknown simulator name and `skip_detect_devices` set to `true` and got 👇 

```sh
[!] undefined method `call' for []:Array
```

🤔 Did you see this at all?

Reply:
Interesting, I did not think to try to run this with an empty list for `matches` AND an empty `default` array. Seems like `call` isn't something that can be invoked on an empty array.

I'll admit, I don't know much about Ruby, but after some googling I have some ideas to try. I'll fiddle with it tonight and see if I can come up with a better solution while testing with your additional scenarios.

Thanks for testing!

Reply:
Ok, this issue should be addressed now. If there is a more eloquent way of writing my solution, please let me know and i'll make the necessary updates. Thank you!

Reply:
Is an invalid name now what you were testing it with? 😇 What simulator name(s) were you using?

Reply:
```rb
# Convert array to lazy enumerable (evaluate map only when needed) 
# Grabs the first unempty evaluated array
Scan.devices = [matches, default].lazy.reject(&:nil?).map { |x|
  arr = x.call
  arr unless arr.empty?
}.reject(&:nil?).first
```

@johndelong I took me some time to understand this code part of the code base (I haven't done anything in here before surprisingly 😛 ). But I _think_ this is the solution we want here ☝️ We will just get rid of the if statement from the original code and add `reject(&:nil?)` before `map`. Thoughts? 😊

Reply:
@joshdholtz Ahhh, I _knew_ there had to be a better way to write this. Very clean! ✨

I tested with an invalid simulator (`iPhone 6sz`), with a valid simulator (`iPhone 6s`), and with `skip_detect_devices` enabled and disabled. This solution checks out ✅ Thank you very much for your review and help. I appreciate it!

TIL about Ruby `.reject(&:nil?)` 😉

Reply:
Glad it works! This was fun to work on 😊 Excited to get this released for you!